assets/repertoire_comparison.Rmd

```diff
@@ -153,14 +153,28 @@ dir.create(abundance_dir)
 dir.create(vfamily_dir)
 
 # Generate one big dataframe from all patient dataframes
-
-df_list = lapply(all_files, read.csv, sep="\t")
-
-df_all <- dplyr::bind_rows(df_list)
+col_select <- c(
+  "sample_id", "subject_id", "sequence_id", "clone_id",
+  "v_call", "d_call", "j_call",
+  "locus",
+  "junction",
+  "pcr_target_locus"
+)
+i <- 0
+df_all <- dplyr::bind_rows(lapply(all_files,
+  function(x) {
+    i <<- i +1
+    message("Loading file #",i,":",x)
+    read_rearrangement(x, col_select=col_select)
+  })
+)
 
 # Remove underscores in these columns
-df_all$subject_id <- sapply(df_all$subject_id, function(x) str_replace(as.character(x), "_", ""))
-df_all$sample_id <- sapply(df_all$sample_id, function(x) str_replace(as.character(x), "_", ""))
+# df_all$subject_id <- sapply(df_all$subject_id, function(x) str_replace(as.character(x), "_", ""))
+df_all$subject_id  <- stringi::stri_replace_first_fixed(df_all$subject_id, "_", "")
+
+# df_all$sample_id <- sapply(df_all$sample_id, function(x) str_replace(as.character(x), "_", ""))
+df_all$sample_id  <- stringi::stri_replace_first_fixed(df_all$sample_id, "_", "")
```

Comment:
```suggestion
df_all$subject_id <- sapply(df_all$subject_id, function(x) str_replace(as.character(x), "_", ""))
df_all$sample_id <- sapply(df_all$sample_id, function(x) str_replace(as.character(x), "_", ""))
```

Reply:
It's not the same function, one is replacing all the `_` and the other one only the first instance. I've hopefully fixed that already in another PR to `dev`